src/genn/backends/hip/backend.cc

```diff
@@ -382,8 +443,15 @@ void Backend::genLazyVariableDynamicAllocation(CodeStream &os, const Type::Resol
 //--------------------------------------------------------------------------
 void Backend::genMakefilePreamble(std::ostream &os) const
 {
-    const std::string architecture = "sm_" + std::to_string(getChosenHIPDevice().major) + std::to_string(getChosenHIPDevice().minor);
-    std::string linkFlags = "--shared -arch " + architecture;
+    std::string architecture;
+#if defined(__HIP_PLATFORM_NVIDIA__)
+    architecture = "sm_" + std::to_string(getChosenHIPDevice().major) + std::to_string(getChosenHIPDevice().minor);
+#elif defined(__HIP_PLATFORM_AMD__)
+    // Get AMD GPU architecture directly from device properties
+    architecture = std::string(getChosenHIPDevice().gcnArchName);
+#endif
+    
+    std::string linkFlags = "--shared --offload-arch=" + architecture + " -fgpu-rdc";
```

Comment:
--offload-arch is getting passed to NVCC when running on NVIDIA which doesn't compile:
```suggestion
    std::string linkFlags = "--shared -fgpu-rdc";
#if defined(__HIP_PLATFORM_NVIDIA__)
    architecture = "sm_" + std::to_string(getChosenHIPDevice().major) + std::to_string(getChosenHIPDevice().minor);
    linkFlags += " -arch " + architecture;
#elif defined(__HIP_PLATFORM_AMD__)
    // Get AMD GPU architecture directly from device properties
    architecture = std::string(getChosenHIPDevice().gcnArchName);
    linkFlags += " --offload-arch=" + architecture;
#endif
```